commit/merkleroot/rmn/client.go

```diff
@@ -19,13 +33,533 @@ type Client interface {
 	ComputeReportSignatures(
 		ctx context.Context,
 		destChain *rmnpb.LaneDest,
-		requestedUpdates []rmnpb.FixedDestLaneUpdateRequest,
+		requestedUpdates []*rmnpb.FixedDestLaneUpdateRequest,
 	) (*ReportSignatures, error)
 }
 
 type ReportSignatures struct {
 	// ReportSignatures are the ECDSA signatures for the lane updates for each node.
+	Signatures []*rmnpb.EcdsaSignature
+	// LaneUpdates are the lane updates for each source chain that we got the signatures for.
 	// NOTE: A signature[i] corresponds to the whole LaneUpdates slice and NOT LaneUpdates[i].
-	Signatures  []rmnpb.EcdsaSignature
-	LaneUpdates []rmnpb.FixedDestLaneUpdate
+	LaneUpdates []*rmnpb.FixedDestLaneUpdate
+}
+
+// PBClient is the base RMN Client implementation.
+type PBClient struct {
```

Comment:
Just a basic naming thing but `PBClient` is not a great name, the RMN client will always use protobuf. I don't think this type really needs to be exported either so it can just be called `client`? The constructor can actually return the private pointer `*client` or the interface type `Client` and still be usable. See top answer [here](https://stackoverflow.com/questions/37952354/golang-public-method-to-private-struct-is-this-have-any-use-case)

Reply:
There's also a section of effective go dedicated to this idea. They call it Generality: https://go.dev/doc/effective_go#generality

Reply:
agree